### PR TITLE
[IMP] mrp: decorate gantt view display name

### PR DESCRIPTION
**Pull request by ajf-odoo**
Show the name of the final product at the beginning of the gantt labels

task: 3977697

See odoo/enterprise#66964

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mrp/models/mrp_workorder.py [hunk 1]
```diff
@@ -271,9 +271,11 @@ def _compute_barcode(self):
             wo.barcode = f"{wo.production_id.name}/{wo.id}"
 
     @api.depends('production_id', 'product_id')
+    @api.depends_context('within_gantt')
     def _compute_display_name(self):
+        fct = (lambda w: f"{w.product_id.name} - {w.production_id.name} - {w.name}") if self.env.context.get('within_gantt') else (lambda w: f"{w.production_id.name} - {w.name}")
         for wo in self:
-            wo.display_name = f"{wo.production_id.name} - {wo.name}"
+            wo.display_name = fct(wo)
```
Comment (amoyaux) [inline addons/mrp/models/mrp_workorder.py:278]:
```suggestion
            wo.display_name = f"{wo.production_id.name} - {wo.name}"
            if self.env.context.get('within_gantt'):
                wo.display_name = f"{w.product_id.name} - {w.production_id.name} - {w.name}"
```
It seems a bit more readable :)

Comment (amoyaux) [reply addons/mrp/models/mrp_workorder.py:278]: I would add a more explicit context like prefix_product

Comment (ajf-odoo) [reply addons/mrp/models/mrp_workorder.py]: I thought it was an elegant pythonic way to avoid n-1 unnecessary tests,
done

**Comment by amoyaux** [comment]
robodoo r+

**Pull request closed without merging**